### PR TITLE
Integrate llvm 20231012

### PR DESCRIPTION
**Pull request by stellaraccident**
Carrying local reverts for:

* https://github.com/openxla/iree/issues/15083 (fixes still being made)
* https://github.com/llvm/llvm-project/pull/67816 while we sync to a corresponding TF version
* https://github.com/llvm/llvm-project/pull/67809, which has already been reverted at HEAD and will drop on its own

Landing with local revert: #15207

### CODE REVIEW

**Review by nicolasvasilache — APPROVED** [review]

**Comment by stellaraccident** [comment]
Bisected the TF failure to:

```
Advance LLVM to e29a253c9ebaded53a823def985364392c4ba4ec: [mlir][tosa][linalg] Apply direct tosa -> linalg Conv2D lowering (#68304) (Jack Frankland on 2023-10-06 17:10:39 -0700) (0 of 62)
```

Will just revert it for now and followup post-integrate.

**Comment by stellaraccident** [comment]
The latest patch fixes the bazel job, and the others looked good before -- so landing.

File: a.out
Comment (MacDue) [inline a.out:1]: Looks like this file was committed by mistake?

Comment (ScottTodd) [reply a.out:1]: Oops, good catch. Let's just delete that...

Comment (ScottTodd) [reply a.out:1]: Aaaand it's gone: https://github.com/openxla/iree/commit/d87a97ec3248b298c2828441b3ebbaaf6c30ad6a

**Comment by stellaraccident** [comment]
Oops. Stuff for a little crazy when trying to splice the unreverts from last week in. Sorry :/

**Pull request merged**